app/services/applications/query.rb

```diff
@@ -11,15 +11,24 @@ def initialize(lead_provider: nil, cohort_start_years: nil, updated_since: nil,
     end
 
     def applications
-      scope = Application
-        .includes(
-          :course,
-          :user,
-          :school,
-          :cohort,
-          :private_childcare_provider,
-          :itt_provider,
-        )
+      # The subquery is an optimization so that we don't have to perform
+      # a separate query for each record as part of Application#previously_funded?
+      scope = all_applications.select(
```

Comment:
I was wondering if there was a better way of modelling this data rather than doing it through a query, as we avoided porting this over initially due to the complexity and cost of the query, maybe we can look into possible options of modelling first before we fallback to this solution?

Reply:
Yeah let's have a chat about this at some point.